lib/TH/THStorage.c

```diff
@@ -43,7 +43,8 @@ TH_API THLongStorage *THLongStorage_newInferSize(THLongStorage *size, ptrdiff_t
   ptrdiff_t i;
   for (i = 0; i < size->size; i++) {
     if (size->data[i] == -1) {
-      THArgCheck(dim_infer == -1, 1, "only one dimension can be inferred");
+      THArgCheck(dim_infer == -1, 1, "asking to infer dimension %ld, but already inferring dimension %ld. \
```

Comment:
How about `asking to infer two dimensions (%ld and %ld), but can only infer one`?